WireAnalytics/Sources/WireAnalytics/AnalyticsEvent.swift

```diff
@@ -0,0 +1,28 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+/// Protocol defining the structure of an analytics event.
+public protocol AnalyticsEvent {
+
+    /// The name of the event.
+    var eventName: String { get }
+
+    /// Additional segmentation data for the event.
+    var segmentation: [SegmentationKeys: String] { get }
```

Comment:
I suggest to make this more type safe:

```swift
protocol AnalyticsValue {

    var analyticsValue: String { get }

}

var segmentation: [SegmentationKey: AnalyticsValue]
```

Or you could go a step further to ensure that keys always have appropriate values:

```swift
protocol SegmentationValue {

    var key: String { get }
    var value: AnalyticsValue { get }

}

struct GroupTypeSegmentationValue: SegmentationValue {

    let key = "is_group"
    let value: AnalyticsValue

    init(value: ConversationType) {
        self.value = value
    }

}
```

Then your segmentation could look like:

```swift
var segmentation: Set<SegmentationValue>
```

Reply:
https://github.com/wireapp/wire-ios/commit/44c3b9deeafb41952c99293d14fbb3c9b817232b